zigup.zig

```diff
@@ -429,6 +448,44 @@ fn listCompilers(allocator: *Allocator) !void {
     }
 }
 
+fn keepOrUnkeepCompiler(allocator: *Allocator, compiler_name_to_keep_or_unkeep: []const u8, keep_or_unkeep: enum { keep, unkeep }) !void {
+    if (mem.eql(u8, "master", compiler_name_to_keep_or_unkeep)) {
+        std.debug.warn("master is always kept, doing nothing\n", .{});
+        return;
+    }
+    const install_dir_string = try getInstallDir(allocator, .{ .create = true });
+    defer allocator.free(install_dir_string);
+
+    // TODO openDirAbsolute in stdlib
+    var install_dir = std.fs.cwd().openDir(install_dir_string, .{ .iterate = true }) catch |e| switch (e) {
```

Comment:
This can just be a normal try, if it fails we want to bubble up an error to the user.